### PR TITLE
Add enumerate services API

### PR DESCRIPTION
**Pull request by Jotheeswaran-Nandagopal**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-python/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Implementation of the enumerate service API, which return the active measurement services
- Added automation test

### Why should this Pull Request be merged?

- This implementation supports the Interactive Approach for the [Measurement Plug-In client API](https://dev.azure.com/ni/DevCentral/_git/ASW?path=/Specs/Test%20Automation%20Frameworks/E2719769_Support%20for%20Measurement%20Plug-In%20client%20APIs/E2719769_Support%20for%20Measurement%20Plug-In%20client%20APIs.md&version=GBmain&_a=preview&anchor=interactive-approach)'s in python 

### What testing has been done?

Existing and new automated tests passed.

### CODE REVIEW

File: packages/service/tests/unit/test_discovery_client.py [hunk 1]
```diff
@@ -319,6 +321,47 @@ def test___discovery_service_exe_unavailable___register_service___raises_file_no
         discovery_client.register_service(_TEST_SERVICE_INFO, _TEST_SERVICE_LOCATION)
 
 
+def test___active_measurements_available___enumerate_services___returns_list_of_measurements(
```
Comment (jayaseelan-james) [inline packages/service/tests/unit/test_discovery_client.py:324]:
```suggestion
def test___registered_measurements___enumerate_services___returns_list_of_measurements(
```

File: packages/service/tests/unit/test_discovery_client.py [hunk 2]
```diff
@@ -24,6 +24,8 @@
     ServiceLocation as GrpcServiceLocation,
     UnregisterServiceRequest,
     UnregisterServiceResponse,
+    EnumerateServicesRequest,
+    EnumerateServicesResponse,
```
Comment (jayaseelan-james) [inline packages/service/tests/unit/test_discovery_client.py:28]: nit: These should be placed above the RegisterServiceRequest following the alphabetical order.

File: packages/service/tests/unit/test_discovery_client.py [hunk 3]
```diff
@@ -319,6 +321,47 @@ def test___discovery_service_exe_unavailable___register_service___raises_file_no
...
+    discovery_service_stub.RegisterService.return_value = RegisterServiceResponse(
+        registration_id="abcd"
+    )
+    discovery_client.register_service(_TEST_SERVICE_INFO, _TEST_SERVICE_LOCATION)
+    discovery_service_stub.RegisterService.assert_called_once()
+    register_service_request: RegisterServiceRequest = (
+        discovery_service_stub.RegisterService.call_args.args[0]
+    )
+    discovery_service_stub.EnumerateServices.return_value = EnumerateServicesResponse(
+        available_services=[register_service_request.service_description]
```
[the remark this answers is not in the document]
Comment (Jotheeswaran-Nandagopal) [reply packages/service/tests/unit/test_discovery_client.py:336]: I referred the `test___service_registered___unregister_service_no_args___sends_request_and_returns_true` test, where they registered the service and then unregistered. So, followed the same to get the service description from the registered services.

[the remark this answers is not in the document]
Comment (Jotheeswaran-Nandagopal) [reply packages/service/tests/unit/test_discovery_client.py]: Updated the logic for mocking by using the test values.

**Review by bkeryan — CHANGES_REQUESTED** [review]

**Review by dixonjoel — APPROVED** [review]

**Review by bkeryan — APPROVED** [review]
Approved with suggestion

File: packages/service/tests/unit/test_discovery_client.py [hunk 4]
```diff
@@ -319,6 +326,50 @@ def test___discovery_service_exe_unavailable___register_service___raises_file_no
         discovery_client.register_service(_TEST_SERVICE_INFO, _TEST_SERVICE_LOCATION)
 
 
+@pytest.mark.parametrize("programming_language", ["Python", "LabVIEW"])
+def test___registered_measurements___enumerate_services___returns_list_of_measurements(
+    discovery_client: DiscoveryClient, discovery_service_stub: Mock, programming_language: str
+):
+    grpc_service_descriptor = GrpcServiceDescriptor(
```
Comment (jayaseelan-james) [inline packages/service/tests/unit/test_discovery_client.py:333]: Can we consider renaming this to something as `expected_service` or `expected_available_service` or `enumerate_services_response`?

**Review by jayaseelan-james — APPROVED** [review]

**Review by SowmyaRavee — APPROVED** [review]

**Pull request merged**